### PR TITLE
Fix clang warnings

### PR DESCRIPTION
**Pull request by makortel**
#### PR description:

This PR fixes the clang warnings reported in https://github.com/cms-sw/cmssw/pull/27983#issuecomment-530972579.

While compiling with clang, it complained about the `alignas` in `SiPixelFedCablingMapGPU.h` I fixed those as well.

#### PR validation:

Code compiles.

### CODE REVIEW

File: RecoLocalCalo/EcalRecAlgos/interface/DeclsForKernels.h [hunk 1]
```diff
@@ -26,10 +26,10 @@
 #include "RecoLocalCalo/EcalRecAlgos/interface/EcalTimeBiasCorrectionsGPU.h"
 #include "RecoLocalCalo/EcalRecAlgos/interface/EcalTimeCalibConstantsGPU.h"
 
-class EcalPulseShape;
+struct EcalPulseShape;
 class EcalSampleMask;
 class EcalTimeBiasCorrections;
-class EcalPulseCovariance;
+struct EcalPulseCovariance;
```
Comment (makortel) [inline RecoLocalCalo/EcalRecAlgos/interface/DeclsForKernels.h]: @vkhristenko Does this (and below) conflict with your ECAL developments?

Comment (vkhristenko) [reply RecoLocalCalo/EcalRecAlgos/interface/DeclsForKernels.h]: if you mean forward decl changing class -> struct - should be fine. i see original one is a struct, therefore definitely should be fine.

Comment (makortel) [reply RecoLocalCalo/EcalRecAlgos/interface/DeclsForKernels.h]: I was more thinking if your PR/developments are touching the same lines (for whatever reason) and thus leading to git merge conflicts.

Comment (vkhristenko) [reply RecoLocalCalo/EcalRecAlgos/interface/DeclsForKernels.h]: Ah, forward decls should be fine (line-wise).

**Comment by fwyzard** [comment]
:construction: Validation running at fu-c2a02-37-01:/data/user/fwyzard/patatrack/validation/run.x9Vn0mBDU3 ...

**Comment by fwyzard** [comment]
## Throughput

plot

**Pull request merged**